api/s2n.h

```diff
@@ -502,6 +502,67 @@ extern int s2n_get_cert_der(const struct s2n_cert *cert, const uint8_t **out_cer
 S2N_API
 extern int s2n_connection_get_peer_cert_chain(const struct s2n_connection *conn, struct s2n_cert_chain_and_key *cert_chain);
 
+/**
+ * Returns the length of DER encoded extension value of the ASN.1 X.509 certificate extension.
+ * 
+ * @param cert A pointer to the s2n_cert object being read.
+ * @param oid_field_in A pointer to a buffer that contains the OID of the X.509 certificate extension to be read.
+ * @param oid_field_in_len represents the length of the input buffer `oid_field_in`.
+ * @param oid_value_len This return value contains the length of DER encoded extension value of the ASN.1 X.509 certificate extension.
+ */
+S2N_API 
+extern int s2n_get_x509_extension_oid_value_length(struct s2n_cert *cert, const uint8_t *oid_field_in,
+                                            const uint32_t oid_field_in_len, uint32_t *oid_value_len);
+
+/**
+ * Returns the DER encoding of an ASN.1 X.509 certificate extension value, it's length and a boolean critical.
+ * 
+ * Note that the memory for `oid_value_out` must be allocated by the caller. The size for this buffer must 
+ * be the maximum value of OID value expected by the caller plus an offset of 2 for the ASN1 X.509 DER encoding sequence.
+ * If the memory allocated is less that required amount, an error is returned. Use `s2n_get_x509_extension_oid_value_length` API 
+ * to determine the size required.
+ * 
+ * @param cert A pointer to the s2n_cert object being read.
+ * @param oid_field_in A pointer to a buffer that contains the OID of the X.509 certificate extension to be read.
+ * @param oid_field_in_len represents the length of the input buffer `oid_field_in`.
+ * @param oid_value_out A pointer to the output buffer which will hold the DER encoding of an ASN.1 X.509 certificate extension value returned. 
+ * @param oid_value_out_len  This value is both an input and output parameter and represents the length of the output buffer `oid_value_out`. 
+ * When used as an input parameter, the caller must use this parameter to convey the maximum length of `oid_value_out`. 
+ * When used as an output parameter, `oid_value_out_len` holds the actual length of the DER encoding of the ASN.1 X.509 certificate extension value returned. 
+ * @param critical This return value contains the boolean value for `critical`.
+ */
+S2N_API 
+extern int s2n_get_x509_extension_oid_value(struct s2n_cert *cert, const uint8_t *oid_field_in, const uint32_t oid_field_in_len,
+                                      uint8_t *oid_value_out, uint32_t *oid_value_out_len, bool *critical);
```

Comment:
Why is oid still in the out param names and in the method names? We are not retrieving oids from the extensions but rather arbitrary binary data.